src/Framework/AspNetCoreAnalyzers/test/Dependencies/AddPackageTests.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Verifiers.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.WebApplicationBuilder.WebApplicationBuilderAnalyzer,
+    Microsoft.AspNetCore.Analyzers.Dependencies.AddPackageFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers.Dependencies;
+
+/// <remarks>
+/// These tests don't assert the fix is applied, since it takes a dependency on the internal
+/// VS-specific `PackageInstallerService`. However, the fixer is invoked in these codepaths
+/// so we can validate that the symbol resolution and checks function correctly.
+/// </remarks>
+public class AddPackagesTest
+{
+    [Fact]
+    public async Task CanFixMissingExtensionMethodForDI()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.{|CS1061:AddOpenApi|}();
+";
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, source);
```

Comment:
Shouldn't there be a namespace changed when the code fix runs?

Reply:
The code comment on this test outlines why it doesn't actually verify the fix.

TL;DR: Although the AspNetPacakgeInstall API is made available via the external access package, the `PackageInstallService` that actually does the work is internal and ships in a netfx package targeted for consumption in Visual Studio.

I experimented with pulling the `AspNetCoreAddPackageCodeAction.TryCreateCodeActionAsync` invocation into a seperate method that we could mock but that would only allow us to see if we made a call out to the external service, not verify that the actual source changed in the way that was intended.

Reply:
I beefed up the tests here so we actually check that the we make a call out to VS to invoke the code action with the correct arguments for the package.